woocommerce/woocommerce-android
# [Jetpack Connection] Handle edge case of custom admin URLs

Closes: #8731 

### Description
This PR adds logic to handle an edge case during the Jetpack Connection from the Application Passwords login state, more details can be found in the ticket and the linked P2.

The PR got big because I needed a new screen that handles the connection, instead of relying on the reusable `WPComWebViewFragment`, which was needed to allow detecting errors in the WebView.

### Testing instructions
**Copied from https://github.com/woocommerce/woocommerce-ios/pull/9354**
- Make sure that WCPay is deactivated on your test store.
- Install and activate [WP Hide](https://wordpress.org/plugins/wp-hide-security-enhancer/) plugin on your store.
- On wp-admin left side bar navigate to WP Hide > Hide Admin. Update a custom URL and enable "Block default Admin Url"
<img width="1087" alt="Screenshot 2023-04-04 at 09 35 33" src="https://user-images.githubusercontent.com/5533851/229679826-54444b80-d040-4eef-a4cd-7d8fbf9dfe97.png">

- On the app, navigate to the Jetpack setup flow by tapping the Jetpack benefit banner on the dashboard screen or go to Menu > Settings > Install Jetpack.
- During the connection step of the setup flow, notice that the connection web view is dismissed and the setup screen should show an error about the connection.

### Images/gif
https://github.com/woocommerce/woocommerce-android/assets/1657201/3817e828-233f-4920-a2f3-4dbfdbeb1162
